### PR TITLE
Do not download unsupported files

### PR DESCRIPTION
**Pull request by vidok**
## Closes https://github.com/elastic/enterprise-search-team/issues/5155

There is no need to download files that can't be processed successfully in the extraction service or ingest pipeline.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

**Review by wangch079 — APPROVED** [review]

File: connectors/sources/sharepoint_online.py [hunk 1]
```diff
@@ -1775,3 +1788,13 @@ async def close(self):
 
     def advanced_rules_validators(self):
         return [SharepointOnlineAdvancedRulesValidator()]
+
+    def is_supported_format(self, filename):
```
Comment (timgrein) [inline connectors/sources/sharepoint_online.py:1792]: nit: that could live in `utils.py` right? `TIKA_SUPPORTED_FILETYPES` also lives in `utils.py`, so would make sense to me to keep things together.

Comment (vidok) [reply connectors/sources/sharepoint_online.py:1792]: Totally agree however,  I'm not sure this is the right moment to move to `utils.py`. The similar code lives in `sharepoint_server.py` and `TIKA_SUPPORTED_FILETYPES ` should not live in `utils.py` either :).

File: connectors/sources/sharepoint_online.py [hunk 2]
```diff
@@ -1775,3 +1788,13 @@ async def close(self):
...
+        if "." not in filename:
+            return False
+
+        attachment_extension = list(os.path.splitext(filename))
```
Comment (navarone-feekery) [inline connectors/sources/sharepoint_online.py:1796]: nit: I think `list` isn't required, because you can use the following `[-1]` on the tuple result.

**Review by navarone-feekery — APPROVED** [review]
LGTM! Left some optional nits.

**Review by navarone-feekery — APPROVED** [review]
Nice! 🚀

**Review by artem-shelkovnikov — APPROVED** [review]
LGTM - I would also add a test that checks if file extension is ALL CAPS to pin the fact that we handle this case too

**Pull request merged**